lesson12/index.html

```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="ja">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <link rel="stylesheet" href="style.css"  />
+    <script src="./main.js" defer></script>
+    <title>TerraceTechフロントエンドエンジニア養成所</title>
+  </head>
+  <body>
+    <div>
+      <input id="js-button" type="button" value="Click!" onclick="onButtonClick()" />
+      <div id="js-loading"></div>
+      <ul id="js-ul"></ul>
+    </div>
+  </body>
+</html>
```

Comment:
IMO
You might want to review your VS code settings as it doesn't seem to end in a new line.

Reply:
Mai-san,
I could not manage to find the reason why the code should end with a new line. 
I appreciate if you can explain the it should leave a blank line at the end.

Reply:
The main reason is that I get a warning on GitHub saying "No new line at end of file".

I don't know much about it, but it seems to follow a standard called POSIX. POSIX is a standard that defines the standard interface for UNIX, a kind of OS, and the system interface with the OS is specified by the specification centered on the functions of the C language. In C, C++ language, if the code does not have a newline at the end of the file, it may not work.

When I was pointed out by an engineer, he said that if there is no line break at the end of a sentence, or if the indent space and tab are not unified, the difference may not be taken well.

In the case of JavaScript, I think there is no reason for it not to work just because there is no line break at the end of a sentence, but it is also a hassle to separate whether or not to break lines for each language, and there is no reason not to add a line break, so I feel like I am adding one.

Below is an article that may be helpful.
https://bottoms-programming.com/archives/indent-tab-space-and-new-line.html#toc6

Reply:
Thank you for the detailed information Mai-san. 
I wasnt aware that not ending with an empty line could cause an error. 
I changed the setting of my VS code.